Code/Framework/AzToolsFramework/Tests/Prefab/Instance/InstanceDeserializationTests.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
+#include <AzToolsFramework/Prefab/PrefabDomUtils.h>
+#include <Prefab/PrefabTestFixture.h>
+
+namespace UnitTest
+{
+    using InstanceDeserializationTest = PrefabTestFixture;
+    using InstanceUniquePointer = AZStd::unique_ptr<AzToolsFramework::Prefab::Instance>;
+
+    static void GenerateDomAndReloadInstantiatedPrefab(InstanceUniquePointer& createdPrefab, InstanceUniquePointer& instantiatedPrefab)
+    {
+        PrefabDom tempDomForStoringAndLoading;
+        PrefabDomUtils::StoreInstanceInPrefabDom(*createdPrefab, tempDomForStoringAndLoading);
+
+        PrefabDomUtils::LoadInstanceFromPrefabDom(
+            *instantiatedPrefab, tempDomForStoringAndLoading, PrefabDomUtils::LoadFlags::UseSelectiveDeserialization);
+    }
+
+    AZStd::pair<InstanceUniquePointer, InstanceUniquePointer> SetupPrefabInstances(
+        const AzToolsFramework::EntityList& entitiesToUseForCreation, PrefabSystemComponent* prefabSystemComponent)
+    {
+        AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> createdPrefab =
+            prefabSystemComponent->CreatePrefab(entitiesToUseForCreation, {}, "test/path");
+        EXPECT_TRUE(createdPrefab != nullptr);
```

Comment:
EXPECT_NE should be used here.
It is able to print the resolved the values of the expressions when a failure occurs, unlike EXPECT_TRUE/EXPECT_FALSE macros which just shows the C++ expression.
i.e if this fails now it would output the message of `createdPrefab != nullptr`, while using `EXPECT_NE(nullptr, createdPrefab)` would output a message such as "expected nullptr, actual <actual address>"

```suggestion
        EXPECT_NE(nullptr, createdPrefab);
```